src/editors/containers/TextEditor/components/SelectImageModal/hooks.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import * as module from './hooks';
+import * as sortUtils from './sortUtils';
+
+export const state = {
+  images: (val) => React.useState(val),
+  highlighted: (val) => React.useState(val),
+  searchString: (val) => React.useState(val),
+  sortFilter: (val) => React.useState(val),
+  loading: (val) => React.useState(val),
+  error: (val) => React.useState(val),
+};
+
+export const getFilteredImages = (images, searchString, sortBy) => {
+  const list = images.filter(img => img.displayName.toLowerCase().includes(searchString.toLowerCase()));
+  switch (sortBy) {
+    case 1:
+      list.sort(sortUtils.dateOldest);
+      break;
+    case 2:
+      list.sort(sortUtils.nameAscending);
+      break;
+    case 3:
+      list.sort(sortUtils.nameDescending);
+      break;
+    default:
+      list.sort(sortUtils.dateNewest);
+      break;
+  }
+  return list;
+};
+
+export const getImg = (imgId, imgList) => imgList.find(img => img.id === imgId);
+
+export const imgHooks = ({ fetchImages, uploadImage, setSelection }) => {
+  const [images, setImages] = module.state.images([]);
+  const [highlighted, setHighlighted] = module.state.highlighted(null);
+  const [searchString, setSearchString] = module.state.searchString('');
+  const [sortFilter, setSortFilter] = module.state.sortFilter(0);
+  const addFileRef = React.useRef();
+  const { loading, startLoading, stopLoading } = module.loadingHooks();
+  const { error, setError } = module.errorHooks();
+
+  React.useEffect(() => {
+    fetchImages({ onSuccess: setImages, stopLoading });
+    stopLoading();
+  }, []);
+
+  return {
+    imgList: getFilteredImages(images, searchString, sortFilter),
+    searchString,
+    setSearchString,
+    sortFilter,
+    setSortFilter,
+    highlighted,
+    setHighlighted,
+    addFileRef,
+    addFileClick: () => addFileRef.current.click(),
+    addFile: (file) => uploadImage({
+      file,
+      startLoading,
+      stopLoading,
+      resetFile: () => {
+        addFileRef.current.value = '';
+      },
+      setError,
+    }),
+    onConfirmSelection: () => setSelection(getImg(highlighted, images)),
+    loading,
+    error,
+    setError,
+  };
+};
+
+export const loadingHooks = () => {
```

Comment:
The way I have loading hooks right now feels kind of janky. Whenever I change it, I seem to break how things work.

Reply:
I return loading in loadingHooks, then return loading as well in imgHooks. I've tried to separate it out so that my index file grabs from loadingHooks directly but this seems to mess up the order of what's called first.